From 59f8d8d352798582444efe6a46769bc65974ab9b Mon Sep 17 00:00:00 2001
From: Jarrett Schultz <jaschultz@microsoft.com>
Date: Tue, 9 Jun 2026 04:40:53 +0000
Subject: [PATCH 2110/2165] HID: Add BUS_SPI support and define HID_SPI_DEVICE
 macro

If connecting a hid_device with bus field indicating BUS_SPI print out
"SPI" in the debug print.

Macro sets the bus field to BUS_SPI and uses arguments to set vendor
product fields.

Signed-off-by: Dmitry Antipov <dmanti@microsoft.com>
Reviewed-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
Signed-off-by: Jingyuan Liang <jingyliang@chromium.org>
(cherry picked from commit 52aac142ca2bc9d1b122c7f8ac033bc895141c13)

diff --git a/drivers/hid/hid-core.c b/drivers/hid/hid-core.c
index 14a06ab435a2378ed859ae6f6140b9c61c3b8dda..b8b57a737390d03994e0834f7e6a28567d50f174 100644
--- a/drivers/hid/hid-core.c
+++ b/drivers/hid/hid-core.c
@@ -2377,6 +2377,9 @@ int hid_connect(struct hid_device *hdev, unsigned int connect_mask)
 	case BUS_I2C:
 		bus = "I2C";
 		break;
+	case BUS_SPI:
+		bus = "SPI";
+		break;
 	case BUS_SDW:
 		bus = "SOUNDWIRE";
 		break;
diff --git a/include/linux/hid.h b/include/linux/hid.h
index 4e2422ec03db7e3078ca29a5f0458b78eb179b60..0aa4cc0a9850d505d34337be3d0a1dbbe190050c 100644
--- a/include/linux/hid.h
+++ b/include/linux/hid.h
@@ -821,6 +821,8 @@ struct hid_descriptor {
 	.bus = BUS_BLUETOOTH, .vendor = (ven), .product = (prod)
 #define HID_I2C_DEVICE(ven, prod)				\
 	.bus = BUS_I2C, .vendor = (ven), .product = (prod)
+#define HID_SPI_DEVICE(ven, prod)				\
+	.bus = BUS_SPI, .vendor = (ven), .product = (prod)
 
 #define HID_REPORT_ID(rep) \
 	.report_type = (rep)
